packages/components/addon/components/hds/code-block/index.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+import { htmlSafe } from '@ember/template';
+import Prism from 'prismjs';
+import { setup } from 'prismjs-glimmer';
+import { guidFor } from '@ember/object/internals';
+
+import 'prismjs/plugins/line-numbers/prism-line-numbers';
+import 'prismjs/plugins/line-highlight/prism-line-highlight';
+
+// Importing language individually because autoloader isn't currently working
+import 'prismjs/components/prism-go';
+import 'prismjs/components/prism-markdown';
+import 'prismjs/components/prism-shell-session';
+
+// languages_path isn't working so autoloader isn't working
+// https://prismjs.com/plugins/autoloader/
+import 'prismjs/plugins/autoloader/prism-autoloader';
+
+setup(Prism);
+// Path is supposed to normally not need to be specified but it's not working either way currently
+// Prism.plugins.autoloader.languages_path = 'prismjs/components';
+
+export default class HdsCodeBlockIndexComponent extends Component {
+  @tracked prismCode = '';
+
+  /**
+   * Generates a unique ID for the code content
+   *
+   * @param preCodeId
+   */
+  preCodeId = 'pre-code-' + guidFor(this);
+
+  /**
+   * @param code
+   * @type {string}
+   * @description code text content for the CodeBlock
+   */
+  get code() {
+    const code = this.args.value;
+
+    assert(
+      '@code for "Hds::CodeBlock" must have a valid value',
+      code !== undefined
+    );
+
+    if (Prism?.plugins?.NormalizeWhitespace) {
+      return Prism.plugins.NormalizeWhitespace.normalize(code);
+    }
+
+    return code;
+  }
+
+  /**
+   * @param language
+   * @type {string}
+   * @default 'javascript'
```

Comment:
Is JavaScript the correct default language? I know it's what we're familiar with, but on the consumer end that's probably closer to the bottom. Open to other opinions, but maybe it should be HCL or bash? Probably something we could poll consumers about.

Reply:
Let's poll consumers.